### ISSUE TITLE
Reimplementation problem

### ISSUE THREAD

**Issue by JJongyn** [maxinge8698/ReptileDistil#1]
Hi, I read your paper very interestingly.

For my reimplementation of your paper, I fine tuned the teacher model, but I only get 49% performance on CoLa. For this, the reptileDistill method only gives 45%. 

Is there something I'm missing?

I add my script below.

python run_glue.py --model_type bert --model_name_or_path bert-base-uncased --task_name CoLA --data_dir ./glue_data/CoLA/ --do_lower_case --max_seq_length 128 --do_train --per_gpu_train_batch_size 32 --per_gpu_eval_batch_size 32 --learning_rate ${lr} --num_train_epochs 5 --output_dir ./model/CoLA/teacher_lr${lr}/ --overwrite_output_dir

**Comment by maxinge8698** [comment]
Thanks for your attention, I think you should try different learning rates from {1e-5, 2e-5, 3e-5, 5e-5} to train a good teacher first, usually a fine-tuned teacher's mcc of 54 or so on the CoLA validation set is normal.

**Comment by JJongyn** [comment]
Thank you for the reply. I was able to achieve 56% teacher performance thanks to you.
However, Against the dev set, the method of the paper achieves its best performance at 46%.

Can you give me a tip?  Is eval_result.txt the correct result for dev-set?

**Comment by JJongyn** [comment]
I was able to re-implement thanks to your help :)

Did you find all the hyperparameters for each dataset? 
If that's right, can each dataset have a different hyperparameter?
